src/Hl7.Fhir.Validation/Schema/IElementSchema.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Hl7.Fhir.ElementModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Hl7.Fhir.Validation.Schema
+{
+    public interface IElementSchema: IAssertion
+    {
+        Uri Id { get; }
+
+        Assertions Members { get; }
+
+        IElementSchema With(IEnumerable<IAssertion> additional);
+
+        Assertions Validate(IEnumerable<ITypedElement> input, ValidationContext vc);
```

Comment:
This is part of `IGroupValidatable`.  So I think `IElementSchema` should derive from `IAssertion` and `IGroupValidatable`